packages/calcite-components/src/components/tile-group/tile-group.tsx

```diff
@@ -52,16 +102,77 @@ export class TileGroup implements InteractiveComponent {
 
   @Element() el: HTMLCalciteTileGroupElement;
 
+  private items: HTMLCalciteTileElement[] = [];
+
+  private slotEl: HTMLSlotElement;
+
   //--------------------------------------------------------------------------
   //
   //  Private Methods
   //
   //--------------------------------------------------------------------------
 
-  private updateTiles = (): void => {
-    this.el.querySelectorAll("calcite-tile").forEach((item) => (item.scale = this.scale));
+  private mutationObserver = createObserver("mutation", () => this.updateItems());
+
+  private setSelectedItems = (emit: boolean, elToMatch?: HTMLCalciteTileElement): void => {
+    if (elToMatch) {
+      this.items?.forEach((el) => {
+        const matchingEl = elToMatch === el;
+        switch (this.selectionMode) {
+          case "multiple":
+            if (matchingEl) {
+              el.selected = !el.selected;
+            }
+            break;
+
+          case "single":
+            el.selected = matchingEl ? !el.selected : false;
+            break;
+
+          case "single-persist":
+            el.selected = !!matchingEl;
+            break;
+        }
+      });
+    }
+
+    this.selectedItems = this.items?.filter((el) => el.selected);
+
+    if (emit) {
+      this.calciteTileGroupSelect.emit();
+    }
+  };
+
+  private setSlotEl = (el: HTMLSlotElement): void => {
+    this.slotEl = el;
   };
 
+  private updateItems = (event?: Event): void => {
```

Comment:
Can you create a separate event handler that calls `updateItems`? I don't think we should be mixing event handlers with item updating methods.